<a name="top"></a>
<div style="width:1000 px">

<div style="float:right; width:98 px; height:98px;">
<img src="https://raw.githubusercontent.com/Unidata/MetPy/main/src/metpy/plots/_static/unidata_150x150.png" alt="Unidata Logo" style="height: 98px;">
</div>

<h1>Update Notebook Material</h1>
<h3>by Unidata</h3>

<div style="clear:both"></div>
</div>

---

This notebook can be used to update material whenever updates are posted.

---

## Running the Update

When you run the following cell, any changes from the workshop GitHub repository will be applied to the material under the directory defined by `gitdir` in your workspace.


In [1]:
repourl = "https://github.com//NCAR/ML_workshop2023/"
branch  = "main"
gitdir  = "ML_workshop2023"

!gitpuller {repourl} {branch} {gitdir}

$ git fetch

$ git reset --mixed

$ git -c user.email=nbgitpuller@nbgitpuller.link -c user.name=nbgitpuller merge -Xours origin/main

Updating 3ea4557..77e23d1

Fast-forward

 .ipynb_checkpoints/00_run_first-checkpoint.ipynb   | 101 +++++

 00_run_first.ipynb                                 |  24 +-

 .../Linear_Models-checkpoint.ipynb                 | 406 ++++++++++-----------

 .../Day1_simple_regression/Linear_Models.ipynb     | 406 ++++++++++-----------

 4 files changed, 506 insertions(+), 431 deletions(-)

 create mode 100644 .ipynb_checkpoints/00_run_first-checkpoint.ipynb



---

## What to expect?

### No Updates Available

If there are no updates to the material, the output from the cell will look something like the following:

~~~bash
$ git fetch

$ git reset --mixed

$ git -c user.email=nbgitpuller@nbgitpuller.link -c user.name=nbgitpuller merge -Xours origin/main

Already up to date.
~~~

### Updates are Available
If there are updates to the material, the output will show the updated files being pulled, for example:

~~~bash
$ git fetch

$ git reset --mixed

$ git -c user.email=nbgitpuller@nbgitpuller.link -c user.name=nbgitpuller merge -Xours origin/main

Updating 3ea4557..77e23d1

Fast-forward

 .ipynb_checkpoints/00_run_first-checkpoint.ipynb   | 101 +++++

 00_run_first.ipynb                                 |  24 +-

 .../Linear_Models-checkpoint.ipynb                 | 406 ++++++++++-----------

 .../Day1_simple_regression/Linear_Models.ipynb     | 406 ++++++++++-----------

 4 files changed, 506 insertions(+), 431 deletions(-)

 create mode 100644 .ipynb_checkpoints/00_run_first-checkpoint.ipynb
~~~

That's it!
Now you have the latest and greatest material.